skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldExecutorService.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.google.common.annotations.VisibleForTesting
+import com.intellij.openapi.components.ServiceManager
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+
+/**
+ * Interface for Skaffold execution service. This service builds and launches Skaffold process from
+ * the given set of standard flags and settings. [DefaultSkaffoldExecutorService] assumes Skaffold
+ * to be already installed and available.
+ */
+interface SkaffoldExecutorService {
+    companion object {
+        val instance
+            get() = ServiceManager.getService(SkaffoldExecutorService::class.java)!!
+    }
+
+    /**
+     * Creates Skaffold command line from the given settings and returns resulting launched
+     * Skaffold system process.
+     *
+     * @param settings Settings with flags supported by Skaffold.
+     * @return [SkaffoldProcess] with resulting process and the command line used to launch it.
+     */
+    fun executeSkaffold(settings: SkaffoldExecutorSettings): SkaffoldProcess
+
+    /**
+     * Execution mode for Skaffold, single run, continuous development, etc.
+     */
+    enum class SkaffoldExecutionMode(val modeFlag: String) {
```

Comment:
I'm not sure the answer but would it be better if this lived in the data class below?

Reply:
I don't have a very strong feeling, both approaches seems fine to me, the only thing it's going to be less visible and the name is going to be very long and deeply nested, making it harder to read. wdyt?

Reply:
I think it makes more sense to be part of the settings

regarding the nested names, maybe promote SkaffoldExecutionSettings to its own class?

Reply:
if you prefer we can refactor this separately from this PR

Reply:
Good idea, let's see how we use this in the next PR, and if this feels better to have it as a separate class and move enum inside of it, we'll refactor it.